Applications/Calculator/Calculator.cpp

```diff
@@ -110,24 +110,23 @@ double Calculator::finish_operation(double argument)
         if ((argument > MAX - m_saved_argument) || (m_saved_argument > MAX + argument)) {
             m_has_error = true;
             return argument;
-        } else {
-            res = m_saved_argument + argument;
         }
+        res = m_saved_argument + argument;
         break;
     case Operation::Subtract:
         if ((argument < MIN + m_saved_argument) || (m_saved_argument < MIN + argument)) {
             m_has_error = true;
-        } else {
-            res = m_saved_argument - argument;
         }
+        res = m_saved_argument - argument;
         break;
     case Operation::Multiply:
-        if ((argument > MAX / m_saved_argument) || (m_saved_argument > MAX / argument)) {
-            m_has_error = true;
-            return argument;
-        } else {
-            res = m_saved_argument * argument;
+        if (argument != 0 || m_saved_argument != 0) {
```

Comment:
If the `argument != 0` is true and `m_saved_argument != 0` is false, then the whole expression evaluates to true and the inner if statement will be executed and will cause a division by zero. You can rearrange your checks and fix this, but isn't there an other way of checking for overflows? For example you can check the sign of the operands and the result. I am not the best person to decide that, maybe @bugaevc can help with that since he is the original author of the application :)

Reply:
Maybe dividing by zero is not that bad if we can display/handle `INF` in over places?